neuro_san/service/http/server/http_server.py

```diff
@@ -265,9 +265,10 @@ def agent_modified(self, agent_name: str, source: AgentStorageSource):
         :param agent_name: name of an agent
         :param source: The AgentStorageSource source of the message
         """
-        # Endpoints configuration has not changed,
-        # so nothing to do here, actually.
-        _ = agent_name
+        agent_service_provider: AsyncAgentServiceProvider = self.allowed_agents.get(agent_name, None)
+        if agent_service_provider is not None:
+            agent_service_provider.reset_service()
+            self.logger.info({}, "Reset service for modified agent %s", agent_name)
 
```

Comment:
When AgentNetwork has been potentially modified, reset it in AsyncAgentServiceProvider
so on the next use AsyncAgentService instance will be reloaded.